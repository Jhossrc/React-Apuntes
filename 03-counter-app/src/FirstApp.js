import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

const FirstApp = ({ greeting, subTitle }) => {
  return (
    <Fragment>
      <h1>CounterApp</h1>
      <hr/>
      <h4> <i>{subTitle}</i> </h4>
      <p>{greeting}</p>
    </Fragment>
  );
};

FirstApp.propTypes = {
    greeting: PropTypes.string.isRequired
}

FirstApp.defaultProps = {
    subTitle : 'Cuenta con nosotros'
}

export default FirstApp;
