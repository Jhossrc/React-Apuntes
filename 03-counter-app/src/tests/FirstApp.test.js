import React from 'react';
import '@testing-library/jest-dom';

import FirstApp from '../FirstApp';
import { shallow } from 'enzyme';

describe('Pruebas en el componente FirstApp', () => {
  test('FirstApp : Debe de mostar <FirstApp/> correctamente', () => {
    const greetingTest = 'Bienvenido a React';
    const wrapper = shallow(<FirstApp greeting={greetingTest} />);

    expect(wrapper).toMatchSnapshot();
  });

  test('FirstApp : Debe de mostar el subTitle enviado por props', () => {
    const greetingTest = 'Bienvenido a React';
    const subTitleTest = 'Aprendiendo Testing';
    const wrapper = shallow(
      <FirstApp greeting={greetingTest} subTitle={subTitleTest} />
    );

    const textSubTitle = wrapper.find('h4 i').text(); // document.querySelector()
    expect(textSubTitle).toBe(subTitleTest);
  });
});
