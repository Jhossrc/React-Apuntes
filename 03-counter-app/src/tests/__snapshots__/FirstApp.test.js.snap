// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente FirstApp FirstApp : Debe de mostar <FirstApp/> correctamente 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <hr />
  <h4>
     
    <i>
      Cuenta con nosotros
    </i>
     
  </h4>
  <p>
    Bienvenido a React
  </p>
</Fragment>
`;
